app/src/main/AndroidManifest.xml

```diff
@@ -4,7 +4,6 @@
     package="com.secuso.privacyfriendlycodescanner.qrscanner">
 
     <uses-permission android:name="android.permission.CAMERA" />
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```

Comment:
This is sadly not possible. MediaStore is only available in API 29+ so for every API level below that we still need external storage permissions

See https://developer.android.com/training/data-storage/shared/media#scoped_storage_enabled

Reply:
Use `<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="28"/>`
instead

Reply:
> MediaStore is only available in API 29+ so for every API level below that we still need external storage permissions
Do you mean what you say or do you mean
> Writing to shared storage with MediaStore without write permission is only available in API 29+
?

Reply:
Well, yeah sorry. MediaStorage does work on lower level APIs but it needs the storage permission to function. My wording was kind of ambiguous... or plain wrong 😄 

> **Scoped storage unavailable**
If your app is used on a device that runs Android 9 or lower, or if your app has temporarily opted out of scoped storage, you must request the READ_EXTERNAL_STORAGE permission to access media files. If you want to modify media files, you must request the WRITE_EXTERNAL_STORAGE permission, as well.
>-- https://developer.android.com/training/data-storage/shared/media#storage-permission

Reply:
I meant: 

MediaStore **without using the permission** is only available in API 29+.

Reply:
Added your suggestion